app/controllers/api/alert_definition_profiles_controller.rb

```diff
@@ -21,5 +23,52 @@ def edit_resource(type, id = nil, data = {})
         raise BadRequestError, "Failed to update alert definition profile - #{err}"
       end
     end
+
+    def get_resource_from_href(href)
+      hrefobj = Href.new(href)
+      resource_search(hrefobj.subject_id, hrefobj.subject, collection_class(hrefobj.subject))
+    end
+
+    def assign_resource(type, id = nil, data = {})
+      profile = resource_search(id, type, collection_class(type))
+      raise BadRequestError, "Must specify either objects or tags for the assignment target" unless data.include?("objects") || data.include?("tags")
+      if data.include?("objects")
+        objects = data["objects"].collect do |ref|
+          get_resource_from_href(ref)
+        end
+        profile.assign_to_objects(objects)
+      end
+
+      if data.include?("tags")
+        data["tags"].each do |tag|
+          profile.assign_to_tags([get_resource_from_href(tag["href"]).classification], tag["class"])
+        end
+      end
+      profile.get_assigned_tos
+    end
+
+    def unassign_resource(type, id = nil, data = {})
+      profile = resource_search(id, type, collection_class(type))
+      raise BadRequestError, "Must specify either objects or tags for the assignment target" unless data.include?("objects") || data.include?("tags")
+
+      if data.include?("objects")
+        objects_to_unassign = data["objects"].collect do |ref|
+          get_resource_from_href(ref)
+        end
+        profile.unassign_objects(objects_to_unassign)
```

Comment:
what happens if the object was not assigned to the tag?

Reply:
Nothing would happen, you won't get an error or anything like it.